src/FiltrationGroupComponent/FiltrationGroupComponent.tsx

```diff
@@ -163,51 +178,62 @@ export class FiltrationGroupComponent extends React.Component<Props> {
 							countChildElement === 1 &&
 							child.classList.contains("kit-filtration-group")
 						) {
-							this.handleHorizontalBracket(child);
+							horizontalBrackets = [
+								...horizontalBrackets,
+								this.handleHorizontalBracket(child)
+							];
 						}
 
 						if (
 							index === ref.childNodes.length - 1 &&
 							child.classList.contains("kit-filtration-group")
 						) {
-							this.handleHorizontalBracket(child, true);
+							horizontalBrackets = [
+								...horizontalBrackets,
+								this.handleHorizontalBracket(child, true)
+							];
 						}
 
 						if (!child.classList.contains("kit-filtration-group")) {
-							this.handleHorizontalBracket(child);
+							horizontalBrackets = [
+								...horizontalBrackets,
+								this.handleHorizontalBracket(child)
+							];
 						}
 					}
 				}
 			});
 		}
+
+		if (horizontalBracket.length !== horizontalBrackets.length) {
+			this.setState({
+				horizontalBracket: horizontalBrackets
+			});
+		}
 	};
 
 	public handleHorizontalBracket = (
 		child: HTMLElement,
 		lastChildGroup?: boolean
 	) => {
-		const label = this.kitFiltrationLabelRef.current;
 		const { minHeight, marginTop } = this;
 
-		const horizontalBracket = document.createElement("span");
-
-		horizontalBracket.className =
-			"kit-filtration-group__label-horizontal-bracket";
-		horizontalBracket.style.top = `${child.offsetTop - 2}px`;
+		let positionTop = child.offsetTop - 2;
 
 		if (lastChildGroup) {
 			const childLabel: HTMLElement | null = child.querySelector(
 				".kit-filtration-group__label"
 			);
 			if (childLabel) {
-				horizontalBracket.style.top = `${child.offsetTop +
+				positionTop =
+					child.offsetTop +
 					childLabel.clientHeight -
 					minHeight -
-					marginTop}px`;
+					marginTop;
 			}
 		}
 
-		label && label.appendChild(horizontalBracket);
+		return { positionTop };
```

Comment:
Вот тут как раз можно просто `number` возвращать.
```suggestion
		return positionTop;
```